lib/zipkin-tracer/faraday/zipkin-tracer.rb

```diff
@@ -10,6 +10,8 @@ def initialize(app, service_name = nil)
     end
 
     def call(env)
+      # Prevent calling this middleware twice
+      return @app.call(env) if !env[:request_headers][b3_headers().values.first].nil?
```

Comment:
- How about printing some warning in the "called twice" case? Silently returning means the user will never know their config is dodgy.
- Code-wise how about making the headers names constants and simply pick one directly, e.g. 
  `return @app.call(env) if env[:request_headers].key?(X-B3-TRACEID)`. 
  The current implementation here will create a new array from the hash values on every single call, even when the config is sane.